ocp_resources/performance_profile.py

```diff
@@ -0,0 +1,93 @@
+
+    def __init__(
+        self,
+        additional_kernel_args=None,
+        cpu=None,
+        globally_disable_irq_load_balancing=None,
+        hugepages=None,
+        machine_config_label=None,
+        machine_config_pool_selector=None,
+        net=None,
+        numa=None,
+        real_time_kernel=None,
+        workload_hints=None,
+        **kwargs,
+    ):
+        """
+        PerformanceProfile object. API reference:
+        https://docs.openshift.com/container-platform/4.14/rest_api/node_apis/performanceprofile-performance-openshift-io-v2.html#performanceprofile-performance-openshift-io-v2
+
+        Args:
+            additional_kernel_args (list, optional): Additional kernel arguments.
+            cpu (dict, required): Set of CPU related parameters.
+            globally_disable_irq_load_balancing (bool, optional): Toggles whether IRQ load balancing will be
+                disabled for the Isolated CPU set.
+            hugepages (dict, optional): Defines a set of huge pages related parameters.
+            machine_config_label (str, optional): Defines the label to add to the MachineConfigs the operator creates.
+            machine_config_pool_selector (str, optional): Defines the MachineConfigPool label to use in the
+                MachineConfigPoolSelector of resources like KubeletConfigs created by the operator.
+            net (dict, optional): Defines a set of network related features.
+            node_selector (str, required): Defines the node label to use in the NodeSelectors of resources like Tuned created by the operator.
+            numa (dict, optional): Defines options related to topology aware affinities.
+            real_time_kernel (dict, optional):  Defines a set of real time kernel related parameters.
+            workload_hints (dict, optional): Defines hints for different types of workloads.
+        """
+
+        super().__init__(**kwargs)
+        self.additional_kernel_args = additional_kernel_args
+        self.cpu = cpu
+        self.globally_disable_irq_load_balancing = globally_disable_irq_load_balancing
+        self.hugepages = hugepages
+        self.machine_config_label = machine_config_label
+        self.machine_config_pool_selector = machine_config_pool_selector
+        self.net = net
+        self.node_selector = kwargs["node_selector"]
+        self.numa = numa
+        self.real_time_kernel = real_time_kernel
+        self.workload_hints = workload_hints
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
+            if not self.cpu:
+                raise ValueError("Passing yaml_file or parameter 'cpu' is required")
+            if not self.node_selector:
+                raise ValueError("Passing yaml_file or parameter 'node_selector' is required")
+
+            manifest_spec = {}
+            manifest_spec["cpu"] = self.cpu
+            manifest_spec["nodeSelector"] = self.node_selector
```

Comment:
define the dict directly ` manifest_spec = {"cpu": ....}`

Reply:
done